zerver/lib/actions.py

```diff
@@ -753,6 +753,17 @@ def do_create_user(
             default_stream_groups=default_stream_groups,
             realm_creation=realm_creation,
         )
+
+    if realm_creation:
+        assert realm.signup_notifications_stream is not None
+        bulk_add_subscriptions(
+            realm, [realm.signup_notifications_stream], [user_profile], acting_user=None
+        )
+
+        from zerver.lib.onboarding import send_initial_realm_messages
+
+        send_initial_realm_messages(realm)
```

Comment:
I'm wondering if this isn't more bug-prone in this order, compared to doing this **before** the `process_new_human_user` call above, given that it does:
```
# Does the processing for a new user account:
# * Subscribes to default/invitation streams
# * Fills in some recent historical messages
# * Notifies other users in realm and Zulip about the signup
# * Deactivates PreregistrationUser objects
def process_new_human_user(
```

But the current ordering is consistent with  how it was being handled inside the `add_new_realm` command so far...

Reply:
Yeah, I was trying to avoid any functional changes in this refactor, and this is the exact same sequencing that happens pre-patch in the registration code path.  (That seemed more likely to be correct than `add_new_realm`, and thus was what I used as a guide).